tools/python/runfiles/runfiles.py

```diff
@@ -161,6 +178,54 @@ def EnvVars(self):
     """
     return self._strategy.EnvVars()
 
+  def CurrentRepository(self, frame = 1):
+    """Returns the canonical name of the caller's Bazel repository.
+
+    For example, this function returns '' (the empty string) when called from
+    the main repository and a string of the form 'rules_python~0.13.0` when
+    called from code in the repository corresponding to the rules_python Bazel
+    module.
+
+    More information about the difference between canonical repository names and
+    the `@repo` part of labels is available at:
+    https://bazel.build/build/bzlmod#repository-names
+
+    NOTE: This function inspects the callstack to determine where in the
+    runfiles the caller is located to determine which repository it came from.
+    This may fail or produce incorrect results depending on who the caller is.
+    Use the `frame` argument to control the stack lookup, or pre-resolve the
+    repository via other means, for example by passing it in from Starlark, to
+    avoid this.
+
+    Args:
+      frame: int; the stack frame to return the repository name for. Defaults to
+          1, the caller of the CurrentRepository function.
+    Returns:
+      The canonical name of the Bazel repository containing the file containing
+      the frame-th caller of this function
+    Raises:
+      ValueError: if the caller cannot be determined or the caller's file path
+          is not contained in the Python runfiles tree
+    """
+    try:
+      caller_path = inspect.getfile(sys._getframe(frame))
+    except ValueError:
```

Comment:
Ah good catch. getframe() says it raises ValueError.
Also catch TypeError, though -- that is what getfile() says it will raise.

Reply:
Done.